frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/dom.rs

```diff
@@ -154,44 +154,56 @@ impl ModuleDom {
                     *state.elem.borrow_mut() = None;
                 }))
                 .apply(clone!(state, sidebar_state, module => move |dom| {
-                    let menu_state = Rc::new(MenuState::new());
-                    let menu_items = match index {
-                        0 => {
-                            vec![
-                                MenuDom::item_edit(menu_state.clone(), state.clone()),
-                                // TODO:
-                                // MenuDom::item_copy(menu_state.clone()),
-                                MenuDom::item_paste(menu_state.clone(), sidebar_state.clone()),
-                            ]
-                        },
-                        _ => {
-                            let mut v = vec![];
-                            if let Some(module) = &*module {
-                                v.push(MenuDom::item_edit(menu_state.clone(), state.clone()));
-                                v.push(MenuDom::item_move_up(menu_state.clone(), state.clone()));
-                                v.push(MenuDom::item_move_down(menu_state.clone(), state.clone()));
-                                v.push(MenuDom::item_duplicate(menu_state.clone(), sidebar_state.clone(), module.id));
-                            }
-                            v.push(MenuDom::item_delete(menu_state.clone(), state.clone()));
-                            if let Some(module) = &*module {
-                                v.push(MenuDom::item_copy(menu_state.clone(), sidebar_state.clone(), module.id));
-                                v.push(MenuDom::item_duplicate_as(menu_state.clone(), sidebar_state.clone(), module));
+                    // If the module is anything other than a DragDrop, otherwise if it is not the
+                    // last module in the list.
+                    if (&*module).is_some() || (index <= total_len - 2 && (&*module).is_none()) {
```

Comment:
Minor nitpick, I don't think that you need to reference and deference, I think that the compiler handles that automatically when calling a method